osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHit.cs

```diff
@@ -67,7 +67,8 @@ public override bool OnPressed(TaikoAction action)
         {
             validKeyPressed = HitActions.Contains(action);
 
-            return UpdateJudgement(true);
+            // Only count this as handled if the new judgement is a hit
+            return UpdateJudgement(true) && Judgements.Last().IsHit;

```

Comment:
is it safe to assume `Judgements` is non-empty at this point?

Reply:
Nope this isn't safe and will fail for early presses.